tests/fuzz/analyzer_fuzzer.go

```diff
@@ -0,0 +1,174 @@
+
+// createRandomConfigFile creates a single fuzzed config file
+func createRandomConfigFile(f *fuzz.ConsumeFuzzer) (string, error) {
+	data, err := f.GetBytes()
+	if err != nil {
+		return "nobytes", err
+	}
+	tmpfile, err := ioutil.TempFile("", "example")
+	if err != nil {
+		return "nofile", err
+	}
+	if _, err := tmpfile.Write(data); err != nil {
+		return "nofile", err
+	}
+	if err := tmpfile.Close(); err != nil {
+		return "nofile", err
+	}
+	return tmpfile.Name(), nil
+}
+
+// createRandomConfigFiles creates a slice of ReaderSources
+func createRandomConfigFiles(f *fuzz.ConsumeFuzzer) ([]local.ReaderSource, error) {
+	var files []local.ReaderSource
+
+	numberOfFiles, err := f.GetInt()
+	if err != nil {
+		return files, err
+	}
+	maxFiles := numberOfFiles % 10
+
+	// Gather test files
+	for i := 0; i < maxFiles; i++ {
+		name, err := f.GetString()
+		if err != nil {
+			return files, err
+		}
+		rBytes, err := f.GetBytes()
+		if err != nil {
+			return files, err
+		}
+		r := bytes.NewReader(rBytes)
+		files = append(files, local.ReaderSource{Name: name, Reader: r})
+	}
+	return files, nil
+}
+
+// runAnalyzer runs the analyzer
+func runAnalyzer(sa *local.SourceAnalyzer) (local.AnalysisResult, error) {
+	prevLogLevel := scope.Processing.GetOutputLevel()
+	scope.Processing.SetOutputLevel(log.NoneLevel)
+	defer scope.Processing.SetOutputLevel(prevLogLevel)
+
+	cancel := make(chan struct{})
+	result, err := sa.Analyze(cancel)
+	if err != nil {
+		return local.AnalysisResult{}, err
+	}
+	return result, err
+}
+
+// FuzzAnalyzer implements the fuzzer
+func FuzzAnalyzer(data []byte) int {
+	f := fuzz.NewConsumer(data)
+	analyzerIndex, err := f.GetInt()
+	if err != nil {
+		return 0
+	}
+	analyzer := availableAnalyzers[analyzerIndex%len(availableAnalyzers)]
+
+	requestedInputsByAnalyzer := make(map[string]map[collection.Name]struct{})
+	analyzerName := analyzer.Metadata().Name
+	cr := func(col collection.Name) {
+		if _, ok := requestedInputsByAnalyzer[analyzerName]; !ok {
+			requestedInputsByAnalyzer[analyzerName] = make(map[collection.Name]struct{})
+		}
+		requestedInputsByAnalyzer[analyzerName][col] = struct{}{}
+	}
+
+	// Mesh config file
+	addMeshConfig, err := f.GetBool()
+	if err != nil {
+		return 0
+	}
+	meshConfigFile := ""
+	if addMeshConfig {
+		meshConfigFile, err = createRandomConfigFile(f)
+		if err != nil {
+			return 0
+		}
+		defer os.Remove(meshConfigFile)
+	}
+
+	// Mesh networks file
+	addMeshNetworks, err := f.GetBool()
+	if err != nil {
+		return 0
+	}
+	meshNetworkFile := ""
+	if addMeshNetworks {
+		meshNetworkFile, err = createRandomConfigFile(f)
+		if err != nil {
+			return 0
+		}
+		defer os.Remove(meshNetworkFile)
+	}
+
+	configFiles, err := createRandomConfigFiles(f)
+	if err != nil {
+		return 0
+	}
+
+	sa := local.NewSourceAnalyzer(schema.MustGet(), analysis.Combine("testCase", analyzer), "", "istio-system", cr, true, 10*time.Second)
+	if addMeshConfig {
+		err = sa.AddFileKubeMeshConfig(meshConfigFile)
```

Comment:
I would be fine with adding a function to directly set from bytes or `MeshConfig` type if file is too slow

Reply:
This fuzzer is not particularly fast, but the bottleneck seems to be `NewSourceAnalyzer`.

Setting from bytes would give a boost to execution speed, but it would still be below 100 execs/s.